---
title: "Hypothesis Testing: Effects of Altering Serotonin Levels for Social Interactions in Mice"
author: "Wellsin Price"
date: "2024/01/15"
mainfont: Monaco
output:
  pdf_document:
    latex_engine: xelatex
    toc: true
    toc_depth: 2
  html_document:
    highlight: tango
    theme: cosmo
    toc: yes
    toc_depth: 2
    toc_float:
      collapsed: false
    df_print: kable
editor_options:
  markdown:
    wrap: sentence
---

```{r, include=FALSE}
# Do not alter // Functional
knitr::opts_chunk$set(echo=TRUE, message=FALSE, warning=FALSE, fig.width=16/2, fig.height=9/2)

# Set seed value for randomness
set.seed(88)
```

# Background {.unnumbered}

Install necessary packages for the analysis.
```{r}
if(!require(tidyverse)){
  install.packages("tidyverse")
}
if(!require(infer)){
  install.packages("infer")
}
if(!require(moderndive)){
  install.packages("moderndive")
}
if(!require(Stat2Data)){
  install.packages("Stat2Data")
}
```

Load the installed packages
```{r}
library(conflicted) # May need to install in cases of package conflicts
library(tidyverse)
library(infer)
library(moderndive)
library(Stat2Data)
```

**Behavioral Effects of Serotonin in Mice Models**

Description: Hypothesis test exploring the effects of altering serotonin levels in social behavior of mice.
Format: A data frame with 48 observations on the following 3 variables.
Study Details: Serotonin is a chemical that influences mood balance in humans.
But how does it affect mice?
Scientists genetically altered mice by "knocking out" the expression of a gene, tryptophan hydroxylase 2 (Tph2), that regulates serotonin production.
With careful breeding, the scientists produced three types of mice that we label as "Minus" for Tph2-/-, "Plus" for Tph2+/+, "Mixed" for Tph2+/-.
The variable Genotype records Minus/Plus/Mixed.
The variable Contacts is the number of social contacts that a mouse had with other mice during an experiment and the variable Sex is "M" for males and "F" for females.
Source: Beis D, Holzwarth K, Flinders M, Bader M, Wohr M, Alenina N., (2015) "Brain serotonin deficiency leads to social communication deficits in mice," Biol.
Lett.
11:20150057.
<http://dx.doi.org/10.1098/rsbl.2015.0057>

```{r}
# Load dataframe
data("MouseBrain")
```

Three Variables:

Contacts Number of social contacts the mouse had during the experiment

Sex F=female or M=male

Genotype Minus, Mixed, or Plus (see description below)

Using hypothesis testing, the following questions will be explored:
1.  Is there a difference in male and female mouse social contacts?
2.  Is there a difference in genotype on social contacts for two conditions using t-tests?
3.  Is there a difference in genotype on social contacts for all three conditions using regression?

# 1: Sex differences in social behavior of mice

## Exploratory Data Analysis (EDA)

```{r}
# Average of social contacts in the dataframe differentiated by sex
MouseBrain %>%
  group_by(Sex) %>%
  summarize(social_mean=mean(Contacts))
```
```{r}
ggplot(MouseBrain, aes(x=Sex, y=Contacts, fill=Sex)) +
  geom_boxplot() +
  labs(x="Number of Contacts by sex")
```

## Null Hypothesis

In order to consider differences as generalizations, a hypothesis test needs to be conducted.
The first step is to declare the hypothesis, starting with the null hypothesis which is the status quo. Thus, the null hypothesis states that there's no difference in mean contacts between the sexes at the population level.
Alternatively, there is a difference at this level between the sexes.

In mathematical notation, it can be described as the following equations:
$$\begin{aligned} H_0:&\mu_{male} = \mu_{female} \\\ \mbox{vs }H_A:& \mu_{male} \neq \mu_{female} \end{aligned}$$ 

$$\begin{aligned} H_0:&\mu_{male} - \mu_{female} = 0 \\\ \mbox{vs }H_A:& \mu_{male} - \mu_{female} \neq 0 \end{aligned}$$

## Testing the hypothesis
The 'infer' package is utilized to conduct the hypothesis test.
### Step 1: Calculate the observed difference
```{r}
# Calculates the observed difference between the sexes
obs_diff_SC_sex <- MouseBrain %>%
  specify(Contacts ~ Sex) %>%
  calculate(stat="diff in means", order = c("F","M"))

obs_diff_SC_sex
```

### Step 2: Generate null distribution of $\delta$

```{r}
# Generates simulated values based on original dataset
SC_in_null_sim <- MouseBrain %>%
  specify(Contacts ~ Sex) %>%
  hypothesize(null="independence") %>%
  generate(reps=5000,type='permute') # generates 5,000 times the original 48 rows
```

### Step 3: Calculate the differences under the null
```{r}
SC_diff_in_null <- SC_in_null_sim %>%
  calculate(stat="diff in means", order=c("F","M"))

SC_diff_in_null %>%
  slice(1:5)
```

### Step 4: Visualize comparison of observed vs null distribution of $\delta$
```{r}
visualize(SC_diff_in_null) +
  shade_p_value(obs_stat=obs_diff_SC_sex, direction="both") +
  labs(x="Difference in mean contacts for males and females",y="Count",
       title="Null distribution of differences in male and female contacts",
       subtitle="Actual difference observed is marked in red"
       )
```

### Step 5: Calculate p-value
```{r}
SC_diff_in_null %>%
  get_pvalue(obs_stat=obs_diff_SC_sex, direction="both")
```

### Step 6: Calculate confidence interval
```{r}
SC_diff_ci <- MouseBrain %>%
  specify(Contacts ~ Sex) %>%
  generate(reps=5000, type="bootstrap") %>%
  calculate(stat="diff in means", order=c("F","M")) %>%
  get_confidence_interval(level=0.95)
SC_diff_ci
```

### Finalize with a t-test
```{r}
t.test(Contacts ~ Sex, var.equal=TRUE, data=MouseBrain)
```

# 2: Effect of Serotonin Genotype on Behavior
As we've seen (and likely may have predicted), the analysis differentiating the sexes proved to not be statistically significant. Thus, the alternative hypothesis is rejected and the null remains. This now turns our attention to the genetics of serotonin: will the expression differentials of Tph2 result in meaningful insights?
In this analysis 'Contacts' represents the outcome variable, and 'Genotype' is the predictor with three levels of expression: "Minus" (-/-), "Mixed" (-/+), and "Plus" (+/+).
Significantly, the Minus condition will be of interest due to the lack of Tph2 expression.
In this analysis, the Minus and Mixed conditions will be compared for significance in social behaviors.

## EDA
```{r}
SC_geno_avg <- MouseBrain %>%
  group_by(Genotype) %>%
  summarize(SC_total=mean(Contacts))

SC_geno_avg
```
```{r}
ggplot(MouseBrain, aes(x=Genotype,y=Contacts)) +
  geom_boxplot(fill="darkslateblue") +
  labs(title="Contact Scores Based on Altered Genotype",
       x="Genotype",y="Contact")
```

## Transforming the Dataset

As we are only exploring Minus and Mixed conditions, the dataset must be transformed to exclude the Plus condition.
```{r}
MouseBrain_minus_mixed <- MouseBrain %>%
  dplyr::filter(Genotype!="Plus")
```

## Stating the Null Hypothesis
The Null Hypothesis in this analysis states that there is no observable difference between Minus and Mixed conditions, whereas the alternative states that there is a statistically significant effect on social contact scores.

## Testing the Hypothesis
### Observed Difference of the Two Genotypes
```{r}
SC_obs_diff_minus_mixed <- MouseBrain_minus_mixed %>%
  specify(Contacts ~ Genotype) %>%
  calculate(stat="diff in means", order=c("Minus", "Mixed"))

SC_obs_diff_minus_mixed
```

### Null Distribution of $\delta$
```{r}
SC_in_null_geno_sim <- MouseBrain_minus_mixed %>%
  specify(Contacts ~ Genotype) %>%
  hypothesize(null="independence") %>%
  generate(reps=5000, type='permute')
```

### Difference Under the Null Distribution
```{r}
SC_diff_in_geno_null <- SC_in_null_geno_sim %>%
  calculate(stat="diff in means", order=c("Minus", "Mixed"))

SC_diff_in_geno_null %>%
  slice(1:5)
```

### Visualize Observed vs Null Distribution Difference
```{r}
visualize(SC_diff_in_geno_null) +
  shade_p_value(obs_stat=SC_obs_diff_minus_mixed, direction = "both") +
  labs(x="Difference in mean contacts between Minus and Mixed", y="Count",
       title="Null Distribution of Minus Mixed Contacts Difference",
       subtitle="Actual Difference Marked in Red")
```

### p-value
```{r}
SC_diff_in_geno_null %>%
  get_pvalue(obs_stat=SC_obs_diff_minus_mixed, direction="both")
```

### Confidence Interval
```{r}
SC_diff_ci_geno <- MouseBrain_minus_mixed %>%
  specify(Contacts ~ Genotype) %>%
  generate(reps=5000, type="bootstrap") %>%
  calculate(stat="diff in means", order=c("Minus", "Mixed")) %>%
  get_confidence_interval(level=0.95)
SC_diff_ci_geno
```

### t.test 
```{r}
t.test(Contacts ~ Genotype, var.equal=TRUE, data=MouseBrain_minus_mixed)
```

### Simple Linear Regression Model
```{r}
# fit regression
mouse_geno_model <- lm(formula=Contacts ~ Genotype, data=MouseBrain)
# Get regression table:
get_regression_table(mouse_geno_model)
```

